contrib/prometheus-cpp-cmake/core/CMakeLists.txt

```diff
@@ -23,6 +23,12 @@ target_link_libraries(core
     $<$<AND:$<BOOL:UNIX>,$<NOT:$<BOOL:APPLE>>>:rt>
 )
 
+if(HAVE_CXX_LIBATOMIC)
```

Comment:
may I ask which symbol is utilized here?

Reply:
libatomic is a sublib of libgcc series to dispatch atomic operations. clang however should generate instrinction directly for most atomic operations as long as the hardware feature flags are set. We do not set `-mcx16` yet for tiflash, but that is something generally avaible for `x86-64-v2` target, which is within the assumption of tiflash. So, if that is the problem, I would prefer adding `-mcx16` on `x86-64` platform instead.

![image](https://user-images.githubusercontent.com/20108837/177033709-8dd80f37-460b-48ef-87ba-1bd2b095f2ae.png)

Reply:
```
if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
  # First check if atomics work without the library.
  check_working_cxx_atomics(HAVE_CXX_ATOMICS_WITHOUT_LIB)
  # If not, check if the library exists, and atomics work with it.
  if(NOT HAVE_CXX_ATOMICS_WITHOUT_LIB)
    check_library_exists(atomic __atomic_load_8 "" HAVE_CXX_LIBATOMIC)
    if(HAVE_CXX_LIBATOMIC)
      list(APPEND CMAKE_REQUIRED_LIBRARIES "atomic")
      check_working_cxx_atomics(HAVE_CXX_ATOMICS_WITH_LIB)
      if(NOT HAVE_CXX_ATOMICS_WITH_LIB)
        message(FATAL_ERROR "Host compiler must support 64-bit std::atomic!")
      endif()
    else()
      message(FATAL_ERROR "Host compiler appears to require libatomic for 64-bit operations, but cannot find it.")
    endif()
  endif()
endif()
```

checking into the implementation, this seems to be a compatible flag for special architectures without even 8-byte atomic load. I think we can skip the check here?

Reply:
I just (carelessly) copy these CMakeLists from the upstream. Nothing will set this variable so it's fine to remove this condition.

The upstream has a another CMakeList, which checked whether the compiler can compile without libatomic. If not, the `HAVE_CXX_LIBATOMIC` will be set. See https://github.com/jupp0r/prometheus-cpp/commit/d0e1056016d4ef1c91317f01e7cce1cb82eb1198

I've removed these lines in the latest commit